config/database.yml

```diff
@@ -24,22 +22,44 @@ defaults: &defaults
     statement_timeout: <%= IdentityConfig.store.database_statement_timeout %>
 
 development:
-  <<: *defaults
+  primary:
+    <<: *defaults
+  primary_replica:
+    <<: *defaults
+    replica: true
 
-test:
+_test: &test
   <<: *defaults
   pool: 10
   checkout_timeout: 10
   database: <%= ENV['POSTGRES_DB'] || "upaya_test#{ENV['TEST_ENV_NUMBER']}" %>
   user: <%= ENV['POSTGRES_USER'] %>
   password: <%= ENV['POSTGRES_PASSWORD'] %>
 
+test:
+  primary:
+    <<: *test
+  primary_replica:
+    <<: *test
+    replica: true
+
 production:
-  <<: *defaults
-  database: <%= IdentityConfig.store.database_name %>
-  username: <%= ProductionDatabaseConfiguration.username %>
-  host: <%= ProductionDatabaseConfiguration.host %>
-  password: <%= ProductionDatabaseConfiguration.password %>
-  pool: <%= ProductionDatabaseConfiguration.pool %>
-  sslmode: 'verify-full'
-  sslrootcert: '/usr/local/share/aws/rds-combined-ca-bundle.pem'
+  primary:
+    <<: *defaults
+    database: <%= IdentityConfig.store.database_name %>
+    username: <%= IdentityConfig.store.database_username %>
+    host: <%= IdentityConfig.store.database_host %>
+    password: <%= IdentityConfig.store.database_password %>
+    pool: <%= IdentityConfig.store.database_pool_idp %>
+    sslmode: 'verify-full'
+    sslrootcert: '/usr/local/share/aws/rds-combined-ca-bundle.pem'
+  primary_replica:
+    <<: *defaults
+    database: <%= IdentityConfig.store.database_name %>
+    username: <%= IdentityConfig.store.database_readonly_username %>
+    host: <%= IdentityConfig.store.database_read_replica_host %>
+    password: <%= IdentityConfig.store.database_readonly_password %>
+    pool: <%= IdentityConfig.store.database_pool_idp %>
+    sslmode: 'verify-full'
+    sslrootcert: '/usr/local/share/aws/rds-combined-ca-bundle.pem'
+    replica: true
```

Comment:
turns out when a connection is set to `replica: true` Rails will block writes also, which is nice

Reply:
Oh sweet that's awesome